### PR TITLE
WT-7639 Alter test_tiered02.py to collect more data before asserting for missing file

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

File: test/suite/test_tiered02.py [hunk 1]
```diff
@@ -90,8 +90,11 @@ def test_tiered(self):
         ds.check()
         self.progress('checkpoint')
         self.session.checkpoint()
+
+        # We need to use sync=on because we check immediately
+        # to see that the files have arrived.
         self.progress('flush_tier')
-        self.session.flush_tier(None)
+        self.session.flush_tier('sync=on')
```
Comment (sueloverso) [inline test/suite/test_tiered02.py:97]: This is very confusing to me. In `api_data.py` the default for `sync` is already `on`. This change should make no difference at all in the execution from the `None` case. Is there something unexpected going on with the config parsing? Can you investigate that?

**Review by sueloverso — COMMENTED** [review]
I need to understand why this change matters. See my comment.

**Comment by ddanderson** [comment]
@sueloverso - I misunderstood that it is on by default.  So this really isn't a fix.

The test fails occasionally (although never for me).  When it fails, it's because an assertion fails in the test - we do a check after every flush_tier to make sure the number of files in the bucket directory has increased.  If not, we assert.  The failures I saw were after the first flush_tier, it complained because 0 wasn't greater than 0, meaning there was nothing copied into the bucketdir.  Since it seems to be a timing issue, I thought maybe there was a race between checking that `flush_state == 0` and decrementing `flush_state`.  I saw the code for the new "flush" flag, and (by mistake) thought that was the problem.

So I guess I'm no closer to solving it.

**Comment by ddanderson** [comment]
I think what I might do is collect more data.  If the test is about to fail in this way, I'll wait a second or two and report if the file has showed up.  I'll still fail in that case, but then we'll know if the file didn't get copied at all, or somehow appeared later.

**Comment by ddanderson** [comment]
@sueloverso, I changed the test to collect info so we'll have an idea if this is a timing problem or some other unreported hard error during the copy/flush.

**Review by sueloverso — APPROVED** [review]
LGTM

**Pull request merged**